Source/Foundation/bsfUtility/Threading/BsTaskScheduler.cpp

```diff
@@ -5,27 +5,26 @@
 
 namespace bs
 {
-	Task::Task(const PrivatelyConstruct& dummy, const String& name, std::function<void()> taskWorker, 
+	Task::Task(const PrivatelyConstruct& dummy, String name, std::function<void()> taskWorker,
 		TaskPriority priority, SPtr<Task> dependency)
-		: mName(name), mPriority(priority), mTaskId(0), mTaskWorker(taskWorker), mTaskDependency(dependency), mState(0)
-		, mParent(nullptr)
+		: mName(std::move(name)), mPriority(priority), mTaskWorker(std::move(taskWorker)), mTaskDependency(std::move(dependency))
 	{
 
 	}
 
-	SPtr<Task> Task::create(const String& name, std::function<void()> taskWorker, TaskPriority priority, SPtr<Task> dependency)
+	SPtr<Task> Task::create(String name, std::function<void()> taskWorker, TaskPriority priority, SPtr<Task> dependency)
 	{
-		return bs_shared_ptr_new<Task>(PrivatelyConstruct(), name, taskWorker, priority, dependency);
+		return bs_shared_ptr_new<Task>(PrivatelyConstruct(), std::move(name), std::move(taskWorker), priority, std::move(dependency));
```

Comment:
I believe we need a separate constructor accepting the rvalues returned by std::move. In this case I'd prefer we just stick with normal references rather than creating more constructors. In fact that's why I have been avoiding passing ownership by value + std::move in most cases, it just creates a daisy chain of extra code and it's easy to mess up by forgetting a move constructor and suddenly your code is doing copies when it should be just passing addresses around.